pkg/clusteragent/clusterchecks/dispatcher_nodes.go

```diff
@@ -33,7 +33,14 @@ func (d *dispatcher) getNodeConfigs(nodeName string) ([]integration.Config, int6
 // processNodeStatus returns configurations dispatched to a given node.
 // returns true if the last configuration change matches the one sent by the node agent.
 func (d *dispatcher) processNodeStatus(nodeName string, status types.NodeStatus) (bool, error) {
+	var upToDate bool
+
 	d.store.Lock()
+	if !d.store.active {
+		// Keep node-agent caches up during warmup phase,
+		// but continue processing the node status.
+		upToDate = true
+	}
```

Comment:
This mechanism is in place to avoid checks not running when a new leader is warming up.
As the DCA needs to count how many nodes are active before dispatching configurations, we rely on the node-agents' cache to keep the checks running.

Reworked the logic and the comments to hopefully make it clearer. WDYT?

Reply:
That's great, thanks 👌